lib/react_client.dart

```diff
@@ -734,8 +746,57 @@ class ReactJsComponentFactoryProxy extends ReactComponentFactoryProxy {
   }
 }
 
+class ReactDartFunctionComponentFactoryProxy extends ReactComponentFactoryProxy
+    with JsBackedMapsComponentFactoryBuilder {
+  /// The ReactJS function used as the type for all [ReactElement]s built by
+  /// this factory.
+  final Function reactFunction;
+
+  /// The name of this function.
+  final String componentName;
+
+  /// The JS component factory used by this factory to build [ReactElement]s.
+  final ReactJsComponentFactory reactComponentFactory;
+
+  ReactDartFunctionComponentFactoryProxy(reactFunction, [String componentName])
+      : this.reactFunction = reactFunction,
+        this.componentName = componentName,
+        this.reactComponentFactory = React.createFactory(reactFunction);
+
+  get type => reactFunction;
+}
+
+/// Creates a function component from the given [dartFunctionComponent] that can be used with React.
+JsFunctionComponent _wrapFunctionComponent(DartFunctionComponent dartFunctionComponent, {String componentName}) {
+  assert(() {
+    componentName ??= getProperty(dartFunctionComponent, 'name');
```

Comment:
I'm confused why this part would only get set in dev mode... what will the `componentName` be in production?

Reply:
This was done in order to reduce any unnecessary additional calls when in production. We do a similar thing with Component2's display name: https://github.com/cleandart/react-dart/blob/5.1.0-wip/lib/react.dart#L608-L615

It will be `Anonymous`:
<img width="1390" alt="Screen Shot 2019-10-24 at 4 14 09 PM" src="https://user-images.githubusercontent.com/41018730/67531964-7ec7a680-f679-11e9-8d57-275a92deb350.png">

That is, unless one is provided via the `componentName`/`displayName` factory arg, then it will use that... which i am thinking in over_react, we will populate with the same string that we would use to generate the UiFactory.